core/src/identity.rs

```diff
@@ -0,0 +1,212 @@
+use serde::{Deserialize, Serialize};
+use signature::keypair::SSIKeyMaterial;
+
+use signature::keypair::Ed25519SSIKeyPair;
+use signature::signer::DIDSigner;
+
+#[derive(Clone, Debug, Copy)]
+pub struct Identity<T> {
```

Comment:
Why make this a generic? DIDResolver is already a trait so I don't see the value here. We can just do:

```
pub struct Identity {
resolver: Box<dyn crate::DIDResolver>,
}
```

Reply:
Assuming functionality is the same, this would prevent a heap allocation by using `T` over `Box<T>`

Reply:
this change returns "the trait `DIDResolver` cannot be made into an object`DIDResolver` cannot be made into an object" error